src/Utilities/Compiler/WellKnownTypeProvider.cs

```diff
@@ -26,14 +26,14 @@ private WellKnownTypeProvider(Compilation compilation)
         {
             Compilation = compilation;
             _fullNameToTypeMap = new ConcurrentDictionary<string, INamedTypeSymbol?>(StringComparer.Ordinal);
-            _referencedAssemblies = new Lazy<ImmutableHashSet<IAssemblySymbol>>(
+            _referencedAssemblies = new Lazy<ImmutableArray<IAssemblySymbol>>(
                 () =>
                 {
                     return ImmutableHashSet.Create<IAssemblySymbol>(
```

Comment:
Avoiding the temporary ImmutableHashSet seems possible. Distinct or regular HashSet would be lighter weight.